source/common/config/filter_json.cc

```diff
@@ -399,5 +399,54 @@ void FilterJson::translateClientSslAuthFilter(
                                       *proto_config.mutable_ip_white_list());
 }
 
+void FilterJson::translateGzipFilter(const Json::Object& json_config,
+                                     envoy::api::v2::filter::http::Gzip& proto_config) {
+  json_config.validateSchema(Json::Schema::GZIP_HTTP_FILTER_SCHEMA);
+
+  JSON_UTIL_SET_BOOL(json_config, proto_config, disable_on_etag);
+  JSON_UTIL_SET_BOOL(json_config, proto_config, disable_on_last_modified);
+
+  JSON_UTIL_SET_INTEGER(json_config, proto_config, content_length);
+  JSON_UTIL_SET_INTEGER(json_config, proto_config, memory_level);
+  JSON_UTIL_SET_INTEGER(json_config, proto_config, window_bits);
+
+  for (const auto& json_content_type : json_config.getStringArray("content_type", true)) {
+    auto* content_type = proto_config.mutable_content_type()->Add();
+    *content_type = json_content_type;
+  }
+
+  for (const auto& json_cache_control : json_config.getStringArray("cache_control", true)) {
+    auto* cache_control = proto_config.mutable_cache_control()->Add();
+    *cache_control = json_cache_control;
+  }
+
+  const auto& json_compression_level = json_config.getString("compression_level", "default");
```

Comment:
technically this is taking a reference to a temporary. I would lose the `&` on the `getString` calls. Same below.